src/views/compute/MigrateWizard.vue

```diff
@@ -36,12 +85,146 @@ export default {
   },
   data () {
     return {
+      loading: true,
+      hosts: [],
+      selectedIndex: null,
+      searchQuery: ''
     }
   },
+  mounted () {
+    this.fetchData()
+  },
   methods: {
+    fetchData () {
+      this.loading = true
+      api('findHostsForMigration', {
+        virtualmachineid: this.resource.id,
+        keyword: this.searchQuery
+      }).then(response => {
+        this.hosts = response.findhostsformigrationresponse.host
+        this.loading = false
+      }).catch(error => {
+        this.$message.error('Failed to load hosts: ' + error)
+      })
+    },
+    submitForm () {
+      this.loading = true
+      api('migrateVirtualMachine', {
+        hostid: this.hosts[this.selectedIndex].id,
+        virtualmachineid: this.resource.id
+      }).then(response => {
+        this.pollActionCompletion(response.migratevirtualmachineresponse.jobid)
+      }).catch(error => {
+        console.error(error)
+        this.$message.error('Failed to migrate host.')
+      })
+    },
+    pollActionCompletion (jobId) {
+      api('queryAsyncJobResult', { jobId }).then(json => {
```

Comment:
@RitchieVincent This seems to be fairly commonly used method/pattern, can you consider refactoring this out to a utility method that we can use as a mixin (see AutogenView or other components for example on using mixins, the utils/mixin.js is one example).

Reply:
I've now created a reusable method which can be used for this in `utils/methods.js`.